app/models/ingest_job.rb

```diff
@@ -272,6 +275,41 @@ def set_study_initialized
     end
   end
 
+  # determine if subsampling needs to be run based on file ingested and current study state
+  def launch_subsample_jobs
+    case self.study_file.file_type
+    when 'Cluster'
+      # only subsample if ingest_cluster was just run, new cluster is > 1K points, and a metadata file is parsed
+      cluster_ingested = self.action.to_sym == :ingest_cluster
+      cluster_needs_subsampling = ClusterGroup.find_by(study_id: self.study.id, study_file_id: self.study_file.id).points > 1000
+      metadata_parsed = self.study.metadata_file.present? && self.study.metadata_file.parsed?
+      if cluster_ingested && cluster_needs_subsampling && metadata_parsed
```

Comment:
the semantics of this are odd. The way variables are named, I'd expect to see something more like:

```
if cluster_needs subsampling
  #initiate subsampling
end
```